### DISCUSSION TITLE
Boilerplate wishlist

### DISCUSSION THREAD

**Discussion by kirtimandwade20** [NeoSOFT-Technologies/rest-java-spring-boot#4] [Ideas]
What should be included in boilerplate
	Unit test cases
	Global exception handler
	Basic Validations handler
	Actuator: Health Checks
	Logger
	Swagger Integrations

**Comment by HarshilNeoSoft** [comment]
Db selections: SQL/NoSQL

**Comment by ranatushar7020** [comment]
Configurations for entity manager if in case we are using complex native queries from single data source.
Configurations for entity manager for connecting multiple data sources(Multiple DBs).
Configurations regarding spring security(OAuth, LDAP, etc).

**Comment by ranatushar7020** [comment]
Configurations for spring batch and service layer template(both using jdbc and jpa)
Configurations for caching mechanisms like ehcache, Redis cache, etc

**Comment by Rimasuthar199** [comment]
Environment and profile setup like Prod - Dev- Staging
